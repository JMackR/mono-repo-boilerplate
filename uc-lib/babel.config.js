module.exports = {
  presets: [["@babel/preset-env", { targets: { node: "current" } }], "module:metro-react-native-babel-preset"],
  plugins: ["@babel/plugin-proposal-class-properties", "inline-json-import"],
  sourceType: "unambiguous",
}
