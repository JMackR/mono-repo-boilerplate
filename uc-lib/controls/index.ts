export * from "./image"
export * from "./button"
export * from "./text"
export * from "./activity-indicator"
export * from "./separator"
export * from "./list"
export * from "./section-list"
export * from "./badge"
export * from "./toggle"
export * from "./login-button"
export * from "./layout"
export * from "./background"
export * from "./grid"
export * from "./touchable"
export * from "./view-block"
export * from "./text-entry"
export * from "./scroll-view"
export * from "./circle"
export * from "./slider"
export * from "./clickable"
export * from "./map"
export * from "./visibility-check"
export * from "./strike-through-text"
// export * from "./stripe-text-entry"
export * from "./input"
