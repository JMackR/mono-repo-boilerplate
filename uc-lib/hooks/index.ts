export * from "./use-infinite-scroll"
export * from "./selectable"
export * from "./screen-provider"
export * from "./loading"
