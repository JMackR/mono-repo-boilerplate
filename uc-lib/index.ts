export * from "./animations"
export * from "./assets"
export * from "./controls"
export * from "./themes"
export * from "./widgets"
export * from "./utilities"
export * from "./hooks"
export * from "./validators"
