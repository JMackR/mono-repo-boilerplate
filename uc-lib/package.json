{
  "name": "redibs-ucl",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "test": "jest --projects test/react test/react-native test/non-react"
  },
  "dependencies": {
    "@types/ramda": "^0.27.14",
    "classnames": "2.2.6",
    "google-map-react": "1.1.7",
    "hoist-non-react-statics": "3.3.2",
    "imask": "6.0.5",
    "react-compound-slider": "2.5.0",
    "react-native-autolink": "3.0.0",
    "react-native-fs": "^2.16.6",
    "zxcvbn": "4.4.2"
  },
  "devDependencies": {},
  "prettier": {
    "printWidth": 125,
    "semi": false,
    "singleQuote": false,
    "trailingComma": "all"
  },
  "jest-junit": {
    "outputDirectory": "../build",
    "ancestorSeparator": " â€º ",
    "classNameTemplate": "{classname}",
    "suiteName": "redibs-redibs-ucl jest tests",
    "suiteNameTemplate": "{filename}",
    "titleTemplate": "{title}",
    "uniqueOutputName": "true"
  }
}
