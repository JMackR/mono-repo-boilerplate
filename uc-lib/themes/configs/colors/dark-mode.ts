import { ColorTheme } from "../../type-defs"
import { deepFreeze } from "shared-lib/utilities/object-utils"

export const _DARK_MODE_THEME_ID = "dark_mode"

export const _DARK_MODE_COLOR_THEME: ColorTheme = deepFreeze({
  identifier: _DARK_MODE_THEME_ID,
  displayName: "Dark mode",
  shade: "dark",
  colors: {
    // Colors
    blurple: "#431AD8",
    grey: "#A3A3A3",
    green: "#00A87E",
    black: "#100D23",
    greenHover: "#43B893",
    greenPressed: "#00986A",
    greenHighlight: "#303D59",
    larchYellow: "#F4BE19",
    larchYellowHover: "#F5D624",
    larchYellowPressed: "#F3A70C",
    glacialBlue: "#3D98D2",
    paintbrushRed: "#E05666",

    // Grayscale
    obsidian: "#F7F7F7",
    basalt: "#E2E2E2",
    granite: "#ABABAB",
    limestone: "#303d59",
    limestoneHover: "#364666",
    limestonePressed: "#262D3F",
    quartz: "#182640",
    crystal: "#001029",
    crystalHover: "#23304B",
    crystalPressed: "#121212",

    // Special
    clear: "rgba(0,0,0,0)",
    disabled: "#182640",
    alwaysDark: "#001029",
    alwaysLight: "#F7F7F7",
    overlay: "rgba(255,255,255,0.6)",
  },
  fontColors: {
    primary: "obsidian",
    primaryAlt: "crystal",
    secondary: "blurple",
    hint: "granite",
    brand: "blurple",
    error: "paintbrushRed",
    alwaysLight: "alwaysLight",
    alwaysDark: "alwaysDark",
  },
  backgroundColors: {
    transparent: "clear",
    primary: "crystal",
    secondary: "quartz",
    tertiary: "limestone",
    alwaysLight: "alwaysLight",
    alwaysDark: "alwaysDark",
    overlay: "overlay",
    error: "paintbrushRed",
    highlight: "larchYellow",
    brand: "blurple",
    unread: "greenHighlight",
    trust: "glacialBlue",
  },
})
