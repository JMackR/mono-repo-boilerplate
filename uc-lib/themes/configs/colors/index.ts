export * from "./light-mode"
export * from "./dark-mode"
