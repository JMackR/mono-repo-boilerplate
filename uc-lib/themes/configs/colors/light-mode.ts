import { ColorTheme } from "../../type-defs"
import { deepFreeze } from "shared-lib/utilities/object-utils"

export const _LIGHT_MODE_THEME_ID = "light_mode"

export const _LIGHT_MODE_COLOR_THEME: ColorTheme = deepFreeze({
  identifier: _LIGHT_MODE_THEME_ID,
  displayName: "Light mode",
  shade: "light",
  colors: {
    // Colors
    blurple: "#431AD8",
    grey: "#A3A3A3",
    green: "#00A87E",
    black: "#100D23",
    greenHover: "#43B893",
    greenPressed: "#00986A",
    greenHighlight: "#DEF3EC",
    larchYellow: "#F4BE19",
    larchYellowHover: "#F5D624",
    larchYellowPressed: "#F3A70C",
    glacialBlue: "#3D98D2",
    paintbrushRed: "#E05666",

    // Grayscale
    obsidian: "#100D23",
    basalt: "#8A8A8A",
    granite: "#ABABAB",
    limestone: "#C3D0DB",
    limestoneHover: "#EEEEEE",
    limestonePressed: "#D0D0D0",
    quartz: "#FAFAFA",
    crystal: "#FFFFFF",
    crystalHover: "#F7F7F7",
    crystalPressed: "#EEEEEE",

    // Special
    clear: "rgba(0,0,0,0)",
    disabled: "#431AD8",
    alwaysDark: "#431AD8",
    alwaysLight: "#FFFFFF",
    overlay: "rgba(18,18,18,0.3)",
  },
  fontColors: {
    primary: "obsidian",
    primaryAlt: "crystal",
    secondary: "grey",
    hint: "granite",
    brand: "blurple",
    error: "paintbrushRed",
    alwaysLight: "alwaysLight",
    alwaysDark: "alwaysDark",
  },
  backgroundColors: {
    transparent: "clear",
    primary: "crystal",
    secondary: "quartz",
    tertiary: "limestone",
    alwaysLight: "alwaysLight",
    alwaysDark: "alwaysDark",
    overlay: "overlay",
    error: "paintbrushRed",
    highlight: "larchYellow",
    brand: "blurple",
    unread: "greenHighlight",
    trust: "glacialBlue",
  },
})
