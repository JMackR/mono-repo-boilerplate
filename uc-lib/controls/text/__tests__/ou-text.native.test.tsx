import { shallow } from "enzyme"
import React from "react"
import { render } from "react-native-testing-library"
import { Text, useColorTheme, useFontTheme } from "uc-lib"
import { _LIGHT_MODE_COLOR_THEME as TEST_THEME } from "uc-lib/themes/configs/colors"
import { _MOBILE_FONT_THEME as TEST_FONTS } from "uc-lib/themes/configs/fonts"

describe("Text Functionality Tests", () => {
  let wrapper: any,
    click = jest.fn()
  beforeEach(() => {
    wrapper = shallow(
      <Text textType="headline1" color="primary" textAlign="center" onPress={click}>
        Test text
      </Text>,
    )
  })
  test("Click calls passed in onClick function", () => {
    wrapper.props().onPress()
    expect(click).toHaveBeenCalledTimes(1)
  })
})

jest.mock("uc-lib/themes/providers/color-theme-context-provider")
useColorTheme.mockReturnValue(TEST_THEME)

jest.mock("uc-lib/themes/providers/font-theme-context-provider")
useFontTheme.mockReturnValue(TEST_FONTS)

describe("Text Snapshot Tests", () => {
  test("Primary large button renders properly", () => {
    const tree = render(<Text>Default text</Text>)

    expect(tree).toMatchSnapshot()
  })
  test("Tertiary large button renders properly", () => {
    const tree = render(
      <Text textType="headline1" color="primary" textAlign="center" onPress={jest.fn()}>
        First text
      </Text>,
    )

    expect(tree).toMatchSnapshot()
  })
  test("Secondary large button renders properly", () => {
    const tree = render(
      <Text textType="secondaryBody1" color="secondary" textAlign="center" onPress={jest.fn()}>
        Second text
      </Text>,
    )
    expect(tree).toMatchSnapshot()
  })

  test("Tertiary text renders properly", () => {
    const tree = render(
      <Text textType="tertiaryBody1" color="error" textAlign="left">
        Second text
      </Text>,
    )
    expect(tree).toMatchSnapshot()
  })

  test("Multiple texts render properly", () => {
    const tree = render(
      <Text>
        <Text textType="primaryBody1" color="primary">
          First text
        </Text>
        <Text textType="secondaryBody1" color="secondary">
          Second text
        </Text>
        <Text textType="tertiaryBody1" color="error">
          Third text
        </Text>
      </Text>,
    )
    expect(tree).toMatchSnapshot()
  })
})
