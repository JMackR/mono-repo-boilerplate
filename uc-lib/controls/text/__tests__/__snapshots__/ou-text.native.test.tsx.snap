// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Snapshot Tests Multiple texts render properly 1`] = `
<Text
  accessibilityLabel="redibs-ucl.text"
  style={
    Array [
      Object {
        "fontFamily": "Mulish-Bold",
        "fontSize": 16,
        "fontWeight": "bold",
        "lineHeight": 20,
      },
      Object {
        "color": "#121212",
        "textAlign": undefined,
        "textDecorationLine": undefined,
      },
    ]
  }
  testID="redibs-ucl.text"
>
  <Text
    accessibilityLabel="redibs-ucl.text"
    style={
      Array [
        Object {
          "fontFamily": "Mulish-Bold",
          "fontSize": 16,
          "fontWeight": "bold",
          "lineHeight": 20,
        },
        Object {
          "color": "#121212",
          "textAlign": undefined,
          "textDecorationLine": undefined,
        },
      ]
    }
    testID="redibs-ucl.text"
  >
    First text
  </Text>
  <Text
    accessibilityLabel="redibs-ucl.text"
    style={
      Array [
        Object {
          "fontFamily": "Mulish-Bold",
          "fontSize": 14,
          "fontWeight": "bold",
          "lineHeight": 16,
        },
        Object {
          "color": "#8A8A8A",
          "textAlign": undefined,
          "textDecorationLine": undefined,
        },
      ]
    }
    testID="redibs-ucl.text"
  >
    Second text
  </Text>
  <Text
    accessibilityLabel="redibs-ucl.text"
    style={
      Array [
        Object {
          "fontFamily": "Mulish-Bold",
          "fontSize": 12,
          "fontWeight": "bold",
          "lineHeight": 14,
        },
        Object {
          "color": "#E05666",
          "textAlign": undefined,
          "textDecorationLine": undefined,
        },
      ]
    }
    testID="redibs-ucl.text"
  >
    Third text
  </Text>
</Text>
`;

exports[`Text Snapshot Tests Primary large button renders properly 1`] = `
<Text
  accessibilityLabel="redibs-ucl.text"
  style={
    Array [
      Object {
        "fontFamily": "Mulish-Bold",
        "fontSize": 16,
        "fontWeight": "bold",
        "lineHeight": 20,
      },
      Object {
        "color": "#121212",
        "textAlign": undefined,
        "textDecorationLine": undefined,
      },
    ]
  }
  testID="redibs-ucl.text"
>
  Default text
</Text>
`;

exports[`Text Snapshot Tests Secondary large button renders properly 1`] = `
<Text
  accessibilityLabel="redibs-ucl.text"
  onPress={[MockFunction]}
  style={
    Array [
      Object {
        "fontFamily": "Mulish-Bold",
        "fontSize": 14,
        "fontWeight": "bold",
        "lineHeight": 16,
      },
      Object {
        "color": "#8A8A8A",
        "textAlign": "center",
        "textDecorationLine": undefined,
      },
    ]
  }
  testID="redibs-ucl.text"
>
  Second text
</Text>
`;

exports[`Text Snapshot Tests Tertiary large button renders properly 1`] = `
<Text
  accessibilityLabel="redibs-ucl.text"
  onPress={[MockFunction]}
  style={
    Array [
      Object {
        "fontFamily": "Mulish-Bold",
        "fontSize": 28,
        "fontWeight": "900",
        "lineHeight": 32,
      },
      Object {
        "color": "#121212",
        "textAlign": "center",
        "textDecorationLine": undefined,
      },
    ]
  }
  testID="redibs-ucl.text"
>
  First text
</Text>
`;

exports[`Text Snapshot Tests Tertiary text renders properly 1`] = `
<Text
  accessibilityLabel="redibs-ucl.text"
  style={
    Array [
      Object {
        "fontFamily": "Mulish-Bold",
        "fontSize": 12,
        "fontWeight": "bold",
        "lineHeight": 14,
      },
      Object {
        "color": "#E05666",
        "textAlign": "left",
        "textDecorationLine": undefined,
      },
    ]
  }
  testID="redibs-ucl.text"
>
  Second text
</Text>
`;
